src/Riok.Mapperly/Descriptors/MappingBuilders/ConvertStaticMethodMappingBuilder.cs

```diff
@@ -0,0 +1,187 @@
+            && SymbolEqualityComparer.Default.Equals(namedSymbol, ctx.Types.DateOnly);
+    }
+
+    private static bool IsDateTimeToTimeOnlyConversion(MappingBuilderContext ctx)
+    {
+        return ctx.Source.SpecialType == SpecialType.System_DateTime
+            && ctx.Types.TimeOnly != null
+            && ctx.Target is INamedTypeSymbol namedSymbol
+            && SymbolEqualityComparer.Default.Equals(namedSymbol, ctx.Types.TimeOnly);
+    }
+
+    private static StaticMethodMapping? TryGetStaticMethodMapping(
+        SymbolAccessor symbolAccessor,
+        List<IMethodSymbol> allMethods,
+        IEnumerable<string> methodNames,
+        ITypeSymbol sourceType,
+        ITypeSymbol targetType,
+        ITypeSymbol nonNullableTargetType,
+        bool targetIsNullable
+    )
+    {
+        // Get all methods with a single parameter whose type is suitable for assignment to the source type,
+        // group them by name,
+        // and convert them to a dictionary whose key is the method name.
+        // The keys in the dictionary are compared case-insensitively to handle possible `Uint` vs `UInt` cases, etc.
+        var allMethodCandidates = allMethods
+            .Where(m => m is { IsStatic: true, ReturnsVoid: false, IsAsync: false, Parameters.Length: 1 })
+            .GroupBy(x => x.Name, x => x, StringComparer.OrdinalIgnoreCase)
+            .ToDictionary(x => x.Key, x => x.ToList(), StringComparer.OrdinalIgnoreCase);
+
+        foreach (var methodName in methodNames)
+        {
+            if (!allMethodCandidates.TryGetValue(methodName, out var candidates))
+                continue;
+
+            if (!targetIsNullable)
+            {
+                var method = candidates.Find(x => symbolAccessor.ValidateSignature(x, nonNullableTargetType, sourceType));
+                if (method != null)
+                    return new StaticMethodMapping(method);
+            }
+            else // currently this branch is never executed because `targetType` is never null. See https://github.com/riok/mapperly/issues/1614
+            {
+                var method = candidates.Find(x => symbolAccessor.ValidateSignature(x, targetType, sourceType));
+                if (method != null)
+                    return new StaticMethodMapping(method);
+            }
+        }
+
+        return null;
+    }
+
+    private static IEnumerable<string> GetTargetStaticMethodNames(MappingBuilderContext ctx)
+    {
+        const string create = "Create";
+        const string from = "From";
+        const string array = "Array";
+
+        if (ctx.Source.IsArrayType(out var arrayType))
+        {
+            yield return $"{create}{from}{arrayType.ElementType.Name}{array}";
+            yield return $"{from}{arrayType.ElementType.Name}{array}";
+
+            if (!arrayType.ElementType.HasKeyword(out var keywordName))
+                yield break;
+
+            yield return $"{create}{from}{keywordName}{array}";
```

Comment:
After reviewing this PR code multiple times, I believe inlining the create, from, and array constants improves readability. While this codebase typically extracts "magic strings" into constants, in this specific case, inlining seems more appropriate as the sole purpose is to concatenate these strings. This approach keeps the code straightforward and easier to follow.

The same applies to `ConvertInstanceMethodMappingBuilder.GetMappingMethodNames`.